adafruit_turtle.py

```diff
@@ -725,504 +866,267 @@ def pencolor(self, c=None):
+    def bgcolor(self, c=None):
         """
-        raise NotImplementedError
-
-    ############################################################################
-    # Filling
-
-    def filling(self):
-        """Not implemented
+        Return or set the background color.
 
-        Return fillstate (True if filling, False else)."""
-        raise NotImplementedError
+        bgcolor()
+            Return the current backgroud color as color specification string.
+            May be used as input to another color/ pencolor/fillcolor call.
 
-    def begin_fill(self):
-        """Not implemented
-
-        To be called just before drawing a shape to be filled."""
-        raise NotImplementedError
+        bgcolor(colorvalue)
+            Set backgroud color to colorvalue, which is a 24-bit integer such as 0xFF0000.
+            The Color class provides the available values:
+            WHITE, BLACK, RED, ORANGE, YELLOW, GREEN, BLUE, PURPLE, PINK
+        """
+        if c is None:
+            return Color.colors[self._bg_color]
+        if c not in Color.colors:
+            raise RuntimeError("Color must be one of the 'Color' class items")
+        old_color = self._bg_color
+        self._fg_palette.make_opaque(old_color)
+        self._bg_color = Color.colors.index(c)
+        self._bg_palette[0] = c
+        self._fg_palette.make_transparent(self._bg_color)
+        self._turtle_palette[0] = c
+        if self._bg_color == self._pencolor:
+            self._turtle_palette.make_transparent(1)
+        else:
+            self._turtle_palette.make_opaque(1)
+        for h in range(self._h):
+            for w in range(self._w):
+                if self._fg_bitmap[w, h] == old_color:
+                    self._fg_bitmap[w, h] = self._bg_color
+        return Color.colors[self._bg_color]
+
+    def set_bgpic(self, file):
+        """
+        Set a picture as background.
 
-    def end_fill(self):
-        """Not implemented
+        set_bgpic(filename)
+            Set backgroud picture using OnDiskBitmap.
+        """
+        self._bg_pic = open(file, 'rb')
+        odb = displayio.OnDiskBitmap(self._bg_pic)
+        self._odb_tilegrid = displayio.TileGrid(odb, pixel_shader=displayio.ColorConverter())
+        self._bg_addon_group.append(self._odb_tilegrid)
+        # centered
+        self._odb_tilegrid.y = ((self._h*self._fg_scale)//2) - (odb.height//2)
+        self._odb_tilegrid.x = ((self._w*self._fg_scale)//2) - (odb.width//2)
+
+    def del_bgpic(self):
+        """
+        Remove the background picture, if any
 
-        Fill the shape drawn after the last call to begin_fill()."""
-        raise NotImplementedError
+        del_bgpic()
+            Remove the picture and close the file
+        """
+        if self._bg_pic is not None:
+            self._bg_addon_group.remove(self._odb_tilegrid)
+            self._odb_tilegrid = None
+            self._bg_pic.close()
+            self._bg_pic = None
```

Comment:
Could you merge the `bgpic` functions into a single function to match https://docs.python.org/3/library/turtle.html#turtle.bgpic ?

Reply:
hello.
yes I can merge the `bgpic` functions, good idea.

for the `changeturtle` function, the original way to change the shape was (IMHO) overcomplicated, with registering of vectored shapes, shape names management etc. This way is simpler, and that covers more use cases : you can use ondiskbitmap, or load an image in memory and use it as shape, or build a bitmap from scratch...

Reply:
`bgpic` functions are now merged and behave like the cpython lib.

Reply:
@dhalbert I added a function not in the initial lib because I don't want to close the door for anybody wanting to implement vectored version for turtle shape. I think it's possible someone wanting to face such a complicated matter do it, but I don't think I would be able to do it myself.
I think you spotted the only function that is not in the original python3 lib.

Reply:
That is fine! I did only a quick glance and only saw that and the `bgpic` function(s).